sdk/src/extras/disk.rs

```diff
@@ -0,0 +1,711 @@
+                    });
+                }
+            };
+
+            if n == 0 {
+                break Ok(());
+            }
+
+            // If the chunk we just read is all zeroes, don't POST it.
+            if !chunk.iter().all(|x| *x == 0) {
+                let encoded = base64::engine::general_purpose::STANDARD.encode(&chunk[0..n]);
+
+                if let Err(e) = senders[i % thread_ct]
+                    .send((offset, encoded, n as u64))
+                    .await
+                {
+                    break Err(e).context("sending chunk to thread failed");
+                }
+            } else {
+                // Bump the progress bar here to make it consistent
+                self.upload_progress.fetch_add(n as u64, Ordering::Relaxed);
+                self.progress_tx
+                    .send_replace(self.upload_progress.load(Ordering::Relaxed));
+            }
+
+            offset += CHUNK_SIZE;
+            i += 1;
+        };
+
+        for tx in senders {
+            drop(tx);
+        }
+
+        read_result?;
+
+        let mut results = Vec::with_capacity(handles.len());
+        for handle in handles {
+            let result = handle.await.context("failed to join upload thread")?;
+            results.push(result);
+        }
+
+        if results.iter().any(|x| x.is_err()) {
+            bail!("one of the upload threads failed");
+        }
+
+        // Stop the bulk write process
+        self.client
+            .disk_bulk_write_import_stop()
+            .project(self.project.clone())
+            .disk(self.disk.clone())
+            .send()
+            .await
+            .context("stopping the bulk write process failed")?;
+
+        // Finalize the disk, optionally making a snapshot
+        self.client
+            .disk_finalize_import()
+            .project(self.project.clone())
+            .disk(self.disk.clone())
+            .body(FinalizeDisk {
+                snapshot_name: self.image_info.as_ref().map(|ii| ii.snapshot.clone()),
+            })
+            .send()
+            .await
+            .context("finalizing the disk failed")?;
+
+        if self.image_info.is_some() {
+            self.create_image().await?;
+        }
+
+        Ok(())
+    }
+
+    async fn create_image(&self) -> Result<()> {
+        let Some(image_info) = &self.image_info else {
+            bail!("no snapshot information provided");
+        };
+
+        // At this point, unwinding is not as important as before. the user
+        // has uploaded their file to a disk and finalized that disk, making
+        // a snapshot out of it. if this step fails, they can always
+        // manually make an image out of the snapshot later and be sure that
+        // the snapshot's contents are the same.
+        let snapshot_id = self
+            .client
+            .snapshot_view()
+            .project(&self.project)
+            .snapshot(NameOrId::Name(image_info.snapshot.clone()))
```

Comment:
`Name` already implements `TryInto<NameOrId>` so you don't need to have the explicit variant here.

Reply:
Good catch, fixed in 7cc0419cc159d5b4a48a8d8c9bec1dddc7a53cb3